src/auth/auth.service.ts

```diff
@@ -1,32 +1,73 @@
 import { Injectable } from '@nestjs/common';
 import { JwtService } from '@nestjs/jwt';
 import { PrismaService } from 'src/prisma/prisma.service';
-import { User } from '@prisma/client';
+import { UserDto } from './utils/types';
+import * as bcrypt from 'bcrypt';
 
 @Injectable()
 export class AuthService {
   constructor(private jwtService: JwtService, private prisma: PrismaService) {}
 
-  signIn(username: string, password: string): Promise<any> {
-    if (username === 'admin' && password === 'admin') {
+  async login(username: string, password: string): Promise<any> {
+    try {
+      const user = await this.prisma.user.findUnique({
+        where: {
+          username: username,
+        },
+      });
+      if (!user) {
+        return Promise.resolve({
+          message: 'Username does not exist',
+        });
+      }
+      const isPasswordValid = await bcrypt.compare(password, user.password);
+      if (!isPasswordValid) {
+        return Promise.resolve({
+          message: 'Password is not valid',
+        });
+      }
       const token = this.jwtService.sign({ username });
       return Promise.resolve({
         message: 'success',
         token: token,
       });
-    } else {
-      return Promise.resolve(null);
+    } catch (error) {
+      return Promise.resolve({
+        message: 'error',
+        error: error,
+      });
     }
   }
 
-  async createUser(data: User): Promise<any> {
```

Comment:
let this be for now

Reply:
let the createUser endpoint be there

Reply:
I changed the end point to more fitting name "register"